lambda/aws-rag-appsync-stepfn-kendra/input_validation/src/lambda.py

```diff
@@ -0,0 +1,64 @@
+#
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+#
+from aws_lambda_powertools.utilities.typing import LambdaContext
+from aws_lambda_powertools import Logger
+
+# !currently useless function!
+
+logger = Logger(service="KENDRA_INGESTION_INPUT_VALIDATION")
+
+def process_files(input_files):
+    files_to_process = []
+    valid = True
+    for i in range(len(input_files)):
+        filename = input_files[i]['name']
+        status = "Unsupported"
+        if isvalid_file_format(filename):
+            status = "Supported"
+            file_to_process = {
+                'status': status,
+                'name': filename
+            }
+            files_to_process.append(file_to_process)
+        else:
+            logger.info("file {filename} extension is currently not supported")
+
+    if not files_to_process:
+        valid = False
+
+    response = {
+        'isValid':valid,
+        'files':files_to_process
+    }
+
+    return response
+
+def isvalid_file_format(file_name: str) -> bool:
+    file_format = ['.pdf','.txt','json','.doc','.docx','.xlsx']
```

Comment:
from https://docs.aws.amazon.com/kendra/latest/dg/index-document-types.html there are other file formats supported by Kendra, is there any reason the pipeline doesn't support them ?

Reply:
Specifically, `ppt`, `rtf`, `csv`, `md`, `html`, and `xml` are listed.